Tests/WATTests/EncoderTests.swift

```diff
@@ -141,6 +145,38 @@ class EncoderTests: XCTestCase {
         #endif
     }
 
+    func testFunctionReferencesProposal() throws {
+        // NOTE: Perform smoke check for function-references proposal here without
+        // bit-to-bit compatibility check with wabt as wabt does not support
+        // function-references proposal yet.
+        for wastFile in Spectest.wastFiles(
+            path: [
+                Spectest.testsuitePath.appendingPathComponent("proposals/function-references")
+            ], include: [], exclude: []
+        ) {
+            print("Checking \(wastFile.path)")
+            var parser = WastParser(
+                try String(contentsOf: wastFile),
+                features: Spectest.deriveFeatureSet(wast: wastFile)
+            )
+            while let directive = try parser.nextDirective() {
+                switch directive {
+                case .module(let directive):
+                    guard case var .text(wat) = directive.source else {
+                        continue
+                    }
+                    _ = wat = wat
+                // TODO: Enable smoke check for encoding
+                // _ = try wat.encode()
```

Comment:
I would like to merge this branch once the WAT/WasmParser implementation passes the Spectest suite. I don't have enough time to work on it this week, so I won't push any further commits, but feel free to ask any questions you may have.

Reply:
Thanks @kateinoigakukun, will look at the commits you pushed and try to focus on the Wat/WasmParser.